api/src/main/java/org/apache/iceberg/expressions/ResidualEvaluator.java

```diff
@@ -89,6 +89,12 @@ public static ResidualEvaluator of(PartitionSpec spec, Expression expr, boolean
     }
   }
 
+  public static ResidualEvaluator of(Expression expr) {
+    // TODO this constructor is used for REST scan planning
```

Comment:
If there isn't enough information, then why create the evaluator at that point? Seems like we should wait until that information is available.

Reply:
Yeah, discussed with @rahil-c this is something that can be late "bound" like Spec/schema.

Reply:
If go with the UnboundFileScanTask idea from here https://github.com/apache/iceberg/pull/11180#discussion_r1796421177 we will also need to bind residual.